### PR TITLE
Adds documentation for the Async Jobs API

### PR DESCRIPTION
**Pull request by jmnsf**
Waiting on https://github.com/getconversio/conversio-app/pull/3745

### CODE REVIEW

**Comment by jmnsf** [comment]
@getconversio/developers ready for review :)

File: source/includes/_async_jobs.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+      "kind": "newsletter-recipients",
+      "status": "pending"
+    },
+    {
+      "id": "57b5aa3b046abfb053d80b57",
+      "kind": "newsletter-recipients",
+      "status": "failed",
+      "startedAt": "2017-01-02T05:33:22.123Z",
+      "completedAt": "2017-01-02T05:33:45.412Z",
+      "error": "Something terrible as happened. Get in touch with support tout de suite!"
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: Spelling mistake here: as -> has

File: source/includes/_async_jobs.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+|**error:**        |**string, optional**|
+|                  |An error message that indicates a problem when processing the job. There is no `result` if this is present.|
+|**result:**       |**string, optional**|
+|                  |The final result from processing this job. What it is depends on job `kind`.|
+|**partnerApp:**   |**string, optional**|
+|                  |The ID of the Partner App that created this Async Job. Included only when authorization is made with the API key.|
+
+## Get Async Job
+
+Returns the requested async job. When using OAuth, only jobs started by the authenticated app can be accessed.
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: Should 'async job' be 'Async Job' just to keep it consistent with the rest of the references to this?

File: source/includes/_async_jobs.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    <li>The given ID does not belong to an existing job</li>
+    <li>The ID belongs to another shop's job</li>
+    <li>The ID belongs to another Partner App's job</li>
+    <li>The job has expired</li>
+  </ul>
+</aside>
+
+### Response Body
+
+Then endpoint returns an object with a `data` key that is itself an object with the Async Job's properties. It will include the Partner App if authorized with an API key. The following keys are returned:
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: Spelling: Then -> The

Comment (RickCraig) [inline source/includes/_async_jobs.md]: I don't know if the first sentence makes 100% sense. Maybe something like:
'The endpoint returns an object with a `data` key. The `data` key is an object with the Async Job's properties.'

File: source/includes/_async_jobs.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+|Key               |Details    |
+|-----------------:|-----------|
+|**id:**           |**string**|
+|                  |The Async Job's ID. You used it to call this endpoint.|
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: The details on this one doesn't sound quite right. Maybe:
'The Async Job's ID. The ID used to call this endpoint.'

This is just a suggestion. It makes it a bit less like you are talking to the reader.

File: source/includes/_async_jobs.md [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+|**kind:**         |**string**|
+|                  |What kind of job this is. Currently only supports "newsletters-recipients".|
+|**status:**       |**string**|
+|                  |The job's status. One of "pending", "done" or "failed".|
+|**startedAt:**    |**string, optional**|
+|                  |When this job started processing. Is an **ISO 8601** encoded date. Can be null if it hasn't started yet.|
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: 'Can be null if it hasn't started yet'. can it be anything else if it hasn't started yet? Maybe: 'Will be null if it hasn't started yet'?

File: source/includes/_async_jobs.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+|**completedAt:**  |**string, optional**|
+|                  |When this job completed. There's a `result` or `error` if this is present. Is an **ISO 8601** encoded date. Can be null if it hasn't finished yet.|
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: Can this be undefined. The reason I ask is, you have stated: 'if this is present'.

Also if you are changing the one above from can -> will then this one needs some love too

File: source/includes/_async_jobs.md [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+|**error:**        |**string, optional**|
+|                  |An error message that indicates a problem when processing the job. There is no `result` if this is present.|
+|**result:**       |**string, optional**|
+|                  |The final result from processing this job. What it is depends on job `kind`.|
```
Comment (RickCraig) [inline source/includes/_async_jobs.md]: I seen this one elsewhere and now that I read it again, I don't know if the sentence 'What it is depends on job `kind`' is proper english. Maybe: 'Content depends on job `kind`'

Comment (RickCraig) [inline source/includes/_async_jobs.md]: or 'Data depends on job `kind`'

File: source/includes/_async_jobs.md [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+
+_OAuth Scopes_: read_async_jobs
+
+<aside class="success">
+  Response 200 (application/json)
+</aside>
+
+### Response Body
+
+Then endpoint returns an object with a `data` key that is an Array with all the async jobs in the shop. It will limit output to the currently authenticated Partner App webhooks if authorized through OAuth.
```
Comment (dlebech) [inline source/includes/_async_jobs.md]: Then -> the

File: source/includes/_async_jobs.md (hunk 3 above)
Comment (jmnsf) [inline source/includes/_async_jobs.md]: Yeah, it's a bit confusing, thanks for the suggestion

**Comment by jmnsf** [comment]
@dlebech @RickCraig thanks for corrections, all fixed ;)

**Pull request merged**